File: Algorithms/p387_First_Unique_Character_in_a_String.cpp
class Solution {
public:
    int firstUniqChar(string s) {
        vector<int> cnt(256);
        for(char c : s){
            cnt[c]++;
        }
        int i = 0;
        for(char c : s){
            if(cnt[c]==1){
                return i;
            }
            i++;
        }
        return -1;
    }
};

File: Algorithms/p387_First_Unique_Character_in_a_String/p387_First_Unique_Character_in_a_String.py
class Solution(object):
    def firstUniqChar(self, s):
        """
        :type s: str
        :rtype: int
        """
        cnt = [0]*256
        for c in s:
            cnt[ord(c)] += 1
        for i,c in enumerate(s):
            if cnt[ord(c)]==1:
                return i
        return -1
